src/components/MovieCreditGroup.tsx

```diff
@@ -0,0 +1,91 @@
+import {useNavigation} from '@react-navigation/native';
+import {StackNavigationProp} from '@react-navigation/stack';
+import React, {FC} from 'react';
+import {StyleSheet, Text, View} from 'react-native';
+import {AppRoute} from '../enums/routes';
+import {useColorTheme} from '../hooks/styles/useColorTheme';
+import {Cast} from '../models/Cast';
+import {Crew} from '../models/Crew';
+import {RootStackNavigatorParams} from '../navigation/RootStackNavigator';
+import MovieCreditCard from './MovieCreditCard';
+
+type MovieScreenProp = StackNavigationProp<
+  RootStackNavigatorParams,
+  AppRoute.FULL_CREDITS
+>;
+
+interface MovieCreditGroupProps {
+  groupName: string;
+  dataSource: Cast[] | Crew[];
+  numberOfItemsToShow?: number;
+}
+
+const MovieCreditGroup: FC<MovieCreditGroupProps> = ({
+  groupName,
+  dataSource,
+  numberOfItemsToShow,
+}) => {
+  const navigation = useNavigation<MovieScreenProp>();
+  const {accentVariantColorForegroundStyle, foregroundVariantStyle} =
+    useColorTheme();
+
+  const getRole = (credit: Cast | Crew): string => {
+    if ('job' in credit) {
+      return credit.job;
+    }
+    return credit.character;
+  };
+
+  const goToFullCreditsList = () => {
+    navigation.navigate(AppRoute.FULL_CREDITS, {
+      dataSource: dataSource,
+      groupName: groupName,
+    });
+  };
+
+  const mapItems = () => {
+    const itemsForRender: JSX.Element[] = dataSource
```

Comment:
Pošto ovu konstantu ne koristimo nigde, možemo odmah i da je vratimo (return bez const)

Reply:
Posto ne prosledjujemo ovom callback-u nikakav parametar, mozemo callback da konvertujemo u najobicniju konstantu `movieCreditCards`